File: Program/LawCreator/test/test/lawcreator/xml/ElementTest.java
package test.lawcreator.xml;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

import pl.wroc.pwr.student.lawcreator.xml.Attribute;
import pl.wroc.pwr.student.lawcreator.xml.AttributeImpl;
import pl.wroc.pwr.student.lawcreator.xml.Element;
import pl.wroc.pwr.student.lawcreator.xml.ElementImpl;

public class ElementTest {
	Element element;

	@Before
	public void setUp() throws Exception {
		element = new ElementImpl();
	}

	@Test
	public void gettingAndSettingFieldsTest() {
		String desc = "desc";
		String name = "name";
		String type = "type";
		
		element.setDescription(desc);
	}
	
	@Test
	public void gettingAndSettingChildrenTest() {
		Element child = new ElementImpl();
		child.setName("name");
		
		element.add(child);
		
		assertEquals(1, element.getElements().size());
		assertEquals("name", ((Element) element.getElements().get(0)).getName());
	}
	
	public void gettingAndSettingAttributesTest() {
		Attribute attribute = new AttributeImpl();
		attribute.setName("name");
		
		element.addAttribute(attribute);
		
		assertEquals(1, element.getAttributes());
		assertEquals(attribute, element.getAttribute("name"));
	}

}

File: Program/LawCreator/bin/test/lawcreator/xml/ElementTest.js
Clazz.declarePackage ("test.lawcreator.xml");
Clazz.load (null, "test.lawcreator.xml.ElementTest", ["pl.wroc.pwr.student.lawcreator.xml.AttributeImpl", "$.ElementImpl"], function () {
c$ = Clazz.decorateAsClass (function () {
this.element = null;
Clazz.instantialize (this, arguments);
}, test.lawcreator.xml, "ElementTest");
Clazz.defineMethod (c$, "setUp", 
function () {
this.element =  new pl.wroc.pwr.student.lawcreator.xml.ElementImpl ();
});
Clazz.defineMethod (c$, "gettingAndSettingFieldsTest", 
function () {
var desc = "desc";
var name = "name";
var type = "type";
this.element.setDescription (desc);
});
Clazz.defineMethod (c$, "gettingAndSettingChildrenTest", 
function () {
var child =  new pl.wroc.pwr.student.lawcreator.xml.ElementImpl ();
child.setName ("name");
this.element.add (child);
org.junit.Assert.assertEquals (1, this.element.getElements ().size ());
org.junit.Assert.assertEquals ("name", (this.element.getElements ().get (0)).getName ());
});
Clazz.defineMethod (c$, "gettingAndSettingAttributesTest", 
function () {
var attribute =  new pl.wroc.pwr.student.lawcreator.xml.AttributeImpl ();
attribute.setName ("name");
this.element.addAttribute (attribute);
org.junit.Assert.assertEquals (new Integer (1), this.element.getAttributes ());
org.junit.Assert.assertEquals (attribute, this.element.getAttribute ("name"));
});
});
